File: src/main/java/com/lukamaret/domain/model/configuration/spotify/SpotifyTokens.java
package com.lukamaret.domain.model.configuration.spotify;

public class SpotifyTokens {

    private String accessToken = "accessToken";
    private String refreshToken = "refreshToken";

    public String getAccessToken() {
        return accessToken;
    }

    public String getRefreshToken() {
        return refreshToken;
    }

    public void setAccessToken(String accessToken) {
        this.accessToken = accessToken;
    }

}

File: src/Domain/Api/SpotifyTokens.php
<?php

namespace App\Domain\Api;

use Symfony\Component\Validator\Constraints as Assert;

class SpotifyTokens
{
    public function __construct(
        #[Assert\NotBlank]
        protected readonly string $accessToken,

        #[Assert\NotBlank]
        protected readonly string $refreshToken,
    ) {
    }

    public function getAccessToken(): string
    {
        return $this->accessToken;
    }

    public function getRefreshToken(): string
    {
        return $this->refreshToken;
    }

    public function equals(object $other): bool
    {
        return $other instanceof self 
            && $other->accessToken === $this->accessToken
            && $other->refreshToken === $this->refreshToken;
    }
}
